### DISCUSSION TITLE
IOS build done, but failed to release the app. Error while archiving ipa

### DISCUSSION THREAD

**Discussion by thangvq95** [codemagic-ci-cd/codemagic-docs#2547] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

i got the issue when i was trying to build the app, code no chages, codemagic workflow no chages, issue just happening today!

### Expected results

App should be created without getting any issue on admin of codemagic.

### Actual results

We are getting issue on app creation

Xcode archive done.                                         271.6s
Failed to build iOS app
Error (Xcode): Undefined symbols:

Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)

Encountered error while archiving for device.

### Build id (optional)

664ebb5c86810854af92ce24

**Comment by leonid-demyanenko** [comment]
We are facing the same issue

**Comment by dtrdic** [comment]
Hello there, can you try the suggestion from this [SO discussion](https://stackoverflow.com/questions/77139621/xcode-15-undefined-symbols-linker-command-failed-with-exit-code-1-use-v-to-s)

@thangvq95 try also checking if [this commit ](https://gitlab.com/doctorworld/care-app/-/commit/9c3b5744573abae142b0baffe3771f73904aefc4) has something to do with the above Linker issue, as from that commit the build started failing

**Reply by thangvq95** [reply]
Hi @dtrdic actually all build for ios is failed today, even if we build the same commit (the one we build success yesterday)
more build id: 664ea73325f28135aa09daf8, 664eeed97d7bb888c6be98d9, ...

**Reply by thangvq95** [reply]
FYI
The build success and failed with same commit and worflow
Build id:
Old 
664da1eb5d5467ab76d74ac8
New (Failed)
664efe264c4a3d25c284cac9

**Reply by dtrdic** [reply]
@thangvq95 can you try removing  `pod repo update` step, maybe it fetched new versions of libraries that led to this error

**Reply by dtrdic** [reply]
In your failed build i can see this two dependencies that have been updated, 
```
image 4.1.7 --> image 4.2.0
recaptcha_enterprise_flutter 18.4.2 --> recaptcha_enterprise_flutter 18.5.0
```
please try to revert to the older versios

**Reply by thangvq95** [reply]
@dtrdic just tried, but still failed: 664f03142898e2289d10e709

**Reply by thangvq95** [reply]
Hi @dtrdic , we ever have uploaded the pubspec.lock file in repo, it auto generate when building the app. these plugins version setup in pubspec is no chagnes.
it's changed from code_magic machine when build the app. i can not update this. btw i don't think it related to dependencies. 

<img width="500" alt="Screenshot 2024-05-23 at 16 24 13" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/13799377/38a8b784-0c00-4f0e-a470-e637e4820e7f">

**Reply by dtrdic** [reply]
@thangvq95 please remove the caret (^) symbol in dependency, from both `image 4.1.7` and `recaptcha_enterprise_flutter 18.4.2` as they are indeed updated during the build. You can check the `Installing dependencies` step in your build and see what versions have been installed.
Try also removing your podfile.lock and the pubspec.lock files from your local project and commit those changes

**Reply by thangvq95** [reply]
Thanks @dtrdic remove ^ to prevent get latest version plugin. issue happening in latest ver 18.5.0 of recaptcha_enterprise_flutter plugin (this version just release today), thank you so much dtrdic!

**Comment by leonid-demyanenko** [comment]
Hello.
It looks like we are having this issue
https://github.com/orgs/codemagic-ci-cd/discussions/2547

We retried previously successful build from the same commit and now it’s
failing

Old
https://codemagic.io/app/64da1f38b92f06247d049f9a/build/664dff6742f24e4165cc3add
New
https://codemagic.io/app/64da1f38b92f06247d049f9a/build/664eec8f64322580e454835f


So it looks like the issue is from Codemagic itself. Same issue is reported
by another project, please check my first link
Looking forward for your reply.

Best regards, Leonid

**Reply by dtrdic** [reply, marked as answer]
Hi @leonid-demyanenko I noticed that some dependencies have been updated when I checked your successful and failed builds. In the failed build, I see the following packages have been updated:
```
archive 3.5.1 --> archive 3.6.0
image 4.1.7 --> image 4.2.0
recaptcha_enterprise_flutter 18.4.2 --> recaptcha_enterprise_flutter 18.5.0
universal_platform 1.0.0+1 --> universal_platform 1.1.0
```
Please try reverting these updates

**Reply by leonid-demyanenko** [reply]
Thank you for the support. It was the cause of the issue @dtrdic 
It's resolved now
